src/Compilers/CSharp/Portable/BoundTree/BoundNodeExtensions.cs

```diff
@@ -94,5 +97,163 @@ private class ContainsAwaitVisitor : BoundTreeWalkerWithStackGuardWithoutRecursi
                 return null;
             }
         }
+
+        public static bool VisitBinaryOperatorInterpolatedString<TArg, TInterpolatedStringType>(
+            this BoundBinaryOperator binary,
+            TArg arg,
+            Func<TInterpolatedStringType, TArg, bool> visitor,
+            Action<BoundBinaryOperator, TArg>? binaryOperatorCallback = null)
+            where TInterpolatedStringType : BoundExpression
+        {
+            Debug.Assert(typeof(TInterpolatedStringType) == typeof(BoundUnconvertedInterpolatedString) || typeof(TInterpolatedStringType) == typeof(BoundInterpolatedString));
+            var stack = ArrayBuilder<BoundBinaryOperator>.GetInstance();
+
+            pushLeftNodes(binary, stack);
+
+            while (stack.TryPop(out BoundBinaryOperator? current))
+            {
+                switch (current.Left)
+                {
+                    case BoundBinaryOperator op:
+                        binaryOperatorCallback?.Invoke(op, arg);
```

Comment:
>binaryOperatorCallback [](http://example.com/codeflow?start=24&length=22)

It looks like the ```binaryOperatorCallback``` is never called for an input that has TInterpolatedStringType on both sides. Is this correct? If so, is this intended and why?